airbyte-integrations/connectors/source-vertica/source_vertica/source.py

```diff
@@ -0,0 +1,173 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+import json
+from datetime import datetime
+from typing import Dict, Generator
+
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.models import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    AirbyteStream,
+    ConfiguredAirbyteCatalog,
+    Status,
+    Type,
+)
+from airbyte_cdk.sources import Source
+
+import vertica_python
+
+
+class SourceVertica(Source):
+    def check(self, logger: AirbyteLogger, config: json) -> AirbyteConnectionStatus:
+        """
+        Tests if the input configuration can be used to successfully connect to the integration
+            e.g: if a provided Stripe API token can be used to connect to the Stripe API.
+
+        :param logger: Logging object to display debug/info/error to the logs
+            (logs will not be accessible via airbyte UI if they are not passed to this logger)
+        :param config: Json object containing the configuration of this source, content of this json is as specified in
+        the properties of the spec.yaml file
+
+        :return: AirbyteConnectionStatus indicating a Success or Failure
+        """
+        try:
+            self.create_connection(config)
+
```

Comment:
Try if you can receive some db metadata as part of the check operation as well.

Reply:
The create_connection() method I have defined returns an object of the Python driver for Vertica. If there is an issue in creating a connection, this will raise an error.

Will it suffice if I do, `connection = self.create_connection(config)`? Although, I don't think it will make much of a diference.

Reply:
I think you should call get_tables() as well to ensure that you have read permissions on the given schema.

Reply:
This is done.